### ISSUE TITLE
Does not respect `exclude` in config or via cli arg

### ISSUE THREAD

**Issue by anentropic** [matthewhughes934/py-unused-deps#29]
I assume because here:

https://github.com/matthewhughes934/py-unused-deps/blob/e7cc68366dffedf37b8dd6125e3b220540880479/unused_deps/main.py#L36-L39

`args.exclude` is used instead of the config that was loaded

but also in the config loader:

https://github.com/matthewhughes934/py-unused-deps/blob/e7cc68366dffedf37b8dd6125e3b220540880479/unused_deps/config.py#L44-L50

any `config_from_file.exclude` value gets overwritten by the default value of `arg.exclude` from here:
https://github.com/matthewhughes934/py-unused-deps/blob/e7cc68366dffedf37b8dd6125e3b220540880479/unused_deps/main.py#L162-L178

There is another problem with that default value - if I attempt to override it from the cli arg I get an error:
```
  File "/Users/pgarner/.pyenv/versions/3.11.5/lib/python3.11/argparse.py", line 1050, in __call__
    items.append(values)
    ^^^^^^^^^^^^
AttributeError: 'tuple' object has no attribute 'append'
```
...the default value is an immutable tuple, so `append` action doesn't work - it needs to be a list

**Comment by matthewhughes934** [comment]
Thanks for the issue, and digging in to find the cause. I think I'll need to rethink the argument parsing and config building a bit, I might just make `--exclude` take a comma separated list, otherwise I think neither `"append"` nor `"extend"` work here as they won't replace the defaults

**Comment by anentropic** [comment]
> Thanks for the issue, and digging in to find the cause.

no prob, these things always get complicated when combinations of options interact with each other! 😅 

I think possibly the simplest way might be to take the arg defaults out the definition of the argparse arg, and instead apply them separately

for example here `for (k, v) in chain(config_from_file.items(), vars(args).items())` it does make sense that `args` overrides `config_from_file`... but only when the args were explicitly given

as a naive suggestion, would:  
`for (k, v) in chain(DEFAULTS.items(), config_from_file.items(), vars(args).items())`
...be approximately the right logic?

I'm sure it would need some extra wrangling to avoid later keys with `None` value overwriting explicit values from earlier in the chain

**Comment by anentropic** [comment]
I've never used it but I think https://docs.python.org/3/library/collections.html#collections.ChainMap is intended for exactly this kind of thing BTW

**Comment by matthewhughes934** [comment]
> I think possibly the simplest way might be to take the arg defaults out the definition of the argparse arg, and instead apply them separately

Yeah, I came to the same realisation soon after trying the other approach, so I went with that. See #30

> I've never used it but I think https://docs.python.org/3/library/collections.html#collections.ChainMap is intended for exactly this kind of thing BTW

:thinking: interesting, I've also not used that one before. I think if it weren't for the filtering that might make the code a bit simpler, but at the same time my use-case doesn't care to keep the existing maps around like that class seems to

**Comment by matthewhughes934** [comment]
This should be fixed with #30
